src/auto/cell_area.rs

```diff
@@ -0,0 +1,341 @@
+    //    unsafe { TODO: call ffi::gtk_cell_area_foreach_alloc() }
+    //}
+
+    fn get_cell_allocation<T: Upcast<Widget>, U: Upcast<CellRenderer>>(&self, context: &CellAreaContext, widget: &T, renderer: &U, cell_area: &Rectangle) -> Rectangle {
+        unsafe {
+            let mut allocation = Rectangle::uninitialized();
+            ffi::gtk_cell_area_get_cell_allocation(self.to_glib_none().0, context.to_glib_none().0, widget.to_glib_none().0, renderer.to_glib_none().0, cell_area.to_glib_none().0, allocation.to_glib_none_mut().0);
+            allocation
+        }
+    }
+
+    fn get_cell_at_position<T: Upcast<Widget>>(&self, context: &CellAreaContext, widget: &T, cell_area: &Rectangle, x: i32, y: i32) -> (CellRenderer, Rectangle) {
+        unsafe {
+            let mut alloc_area = Rectangle::uninitialized();
+            let ret = from_glib_none(ffi::gtk_cell_area_get_cell_at_position(self.to_glib_none().0, context.to_glib_none().0, widget.to_glib_none().0, cell_area.to_glib_none().0, x, y, alloc_area.to_glib_none_mut().0));
+            (ret, alloc_area)
+        }
+    }
+
+    fn get_current_path_string(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gtk_cell_area_get_current_path_string(self.to_glib_none().0))
+        }
+    }
+
+    fn get_edit_widget(&self) -> Option<CellEditable> {
+        unsafe {
+            from_glib_none(ffi::gtk_cell_area_get_edit_widget(self.to_glib_none().0))
+        }
+    }
+
+    fn get_edited_cell(&self) -> Option<CellRenderer> {
+        unsafe {
+            from_glib_none(ffi::gtk_cell_area_get_edited_cell(self.to_glib_none().0))
+        }
+    }
+
+    fn get_focus_cell(&self) -> Option<CellRenderer> {
+        unsafe {
+            from_glib_none(ffi::gtk_cell_area_get_focus_cell(self.to_glib_none().0))
+        }
+    }
+
+    fn get_focus_from_sibling<T: Upcast<CellRenderer>>(&self, renderer: &T) -> Option<CellRenderer> {
+        unsafe {
+            from_glib_none(ffi::gtk_cell_area_get_focus_from_sibling(self.to_glib_none().0, renderer.to_glib_none().0))
+        }
+    }
+
+    fn get_focus_siblings<T: Upcast<CellRenderer>>(&self, renderer: &T) -> Vec<CellRenderer> {
+        unsafe {
+            FromGlibPtrContainer::from_glib_none(ffi::gtk_cell_area_get_focus_siblings(self.to_glib_none().0, renderer.to_glib_none().0))
+        }
+    }
+
+    fn get_preferred_height<T: Upcast<Widget>>(&self, context: &CellAreaContext, widget: &T) -> (i32, i32) {
+        unsafe {
+            let mut minimum_height = mem::uninitialized();
+            let mut natural_height = mem::uninitialized();
+            ffi::gtk_cell_area_get_preferred_height(self.to_glib_none().0, context.to_glib_none().0, widget.to_glib_none().0, &mut minimum_height, &mut natural_height);
```

Comment:
I don't see why you'd want that. I've never seen such a style guideline in any Rust project.

This isn't ANSI C where the declarations are grouped at the top. There doesn't seem to be a logical reason to separate variable initialization and use here (I can see blank lines between logically separate blocks of code). The ’let` keyword should be prominent enough.